lib/spack/spack/cmd/env.py

```diff
@@ -89,26 +91,39 @@ def env_activate(args):
         return 1
 
     if ev.exists(env) and not args.dir:
-        spack_env = ev.root(env)
+        env_path = ev.root(env)
         short_name = env
         env_prompt = '[%s]' % env
 
     elif ev.is_env_dir(env):
-        spack_env = os.path.abspath(env)
-        short_name = os.path.basename(os.path.abspath(env))
+        env_path = os.path.abspath(env)
+        short_name = os.path.basename(env_path)
         env_prompt = '[%s]' % short_name
 
     else:
         tty.die("No such environment: '%s'" % env)
 
-    if spack_env == os.environ.get('SPACK_ENV'):
-        tty.debug("Environment %s is already active" % args.activate_env)
-        return
+    # Shell commands
+    cmds = ''
+
+    # Deactivate the current environment first.  We don't properly support
+    # stacked environments at the moment.
+    if ev.active_environment() is not None:
```

Comment:
The only potential issue here is that `-e` can override currently active env. We should probably just error out for stuff like this

```
spack -e ... env active ...
spack -e ... env deactivate
```

or not @becker33?

Reply:
This is resolved in #25625